components/storage/utils.ts

```diff
@@ -1,110 +1,158 @@
-import { ServiceConfig } from "@/lib/storage-types"
+import {FormQuestions, SelectedAnswers, ServiceConfig,} from "@/lib/storage-types"
 
 /**
- * Extracts all unique feature names from an array of services.
- * @param services Array of ServiceConfig objects
- * @returns Array of unique feature names (strings)
+ * Determines whether text should be white or black based on the background color
+ * Uses WCAG contrast guidelines for accessibility
+ *
+ * @param {string} backgroundColor - The background color (CSS class or hex)
+ * @returns {string} - 'text-white' or 'text-black'
  */
-export function getAllUniqueFeatureNames(services: ServiceConfig[]): string[] {
-  const uniqueFeatureNames = new Set<string>()
-  services.forEach((service) => {
-    service.features?.forEach((feature) => {
-      uniqueFeatureNames.add(feature.name)
-    })
-  })
-  return Array.from(uniqueFeatureNames)
+export const getTextColorForBackground = (backgroundColor: string): string => {
+  const colorMap: Record<string, string> = {
+    "bg-keppel-500": "text-black",
+    "bg-keppel-600": "text-white",
+    "bg-keppel-700": "text-white",
+    "bg-sunglow-400": "text-black",
+    "bg-sunglow-200": "text-black",
+    "bg-red-university": "text-white",
+    "bg-red-500": "text-white",
+    "bg-red-600": "text-white",
+    "bg-amber-600": "text-white",
+    "bg-amber-500": "text-white",
+    "bg-cyan-500": "text-white",
+    "bg-purple-900": "text-white",
+    "bg-purple-500": "text-white",
+    "bg-blue-500": "text-white",
+    "bg-blue-600": "text-white",
+    "bg-pink-500": "text-white",
+    "bg-neutral-800": "text-white",
+    "bg-neutral-900": "text-white",
+    "bg-neutral-200": "text-black",
+    "bg-neutral-300": "text-black",
+    "bg-neutral-400": "text-white",
+    "bg-neutral-500": "text-white",
+    "bg-neutral-600": "text-white",
+    "bg-neutral-700": "text-white",
+  }
+  return colorMap[backgroundColor] || "text-black"
 }
 
 /**
- * Sorts features or feature names according to the order in questionsConfig, then alphabetically.
- * Accepts an array of features ({ name: string }) or feature names (string[]), and a questionsConfig array.
- * @template T - Either { name: string } or string
- * @param features Array of features or feature names
- * @param questionsConfig Array of question configs with affected_feature
- * @returns Sorted array of features or feature names
+ * Maps feature values to badge background colors based on the storage-types color map
  */
-export function sortFeatures<T extends { name: string } | string>(
-  features: T[],
-  questionsConfig: { affected_feature: string }[]
-): T[] {
-  if (!features) return []
-  if (!questionsConfig) {
-    return features.slice().sort((a, b) => {
-      const aName = typeof a === "string" ? a : a.name
-      const bName = typeof b === "string" ? b : b.name
-      return aName.localeCompare(bName)
-    })
+export const getBadgeBackgroundColor = (
```

Comment:
this is also duplicated in the Badge compoennt - but i think it now only has hardcoded values for storage comparison values... so could stay here and be removed from Badge?